translation(JS): Math min and max (#136)
webdoky/content: files/uk/web/javascript/reference/global_objects/math/max/index.md

---
title: Math.max()
slug: Web/JavaScript/Reference/Global_Objects/Math/max
tags:
  - JavaScript
  - Math
  - Method
  - Maximum
  - Reference
  - Largest
  - Largest Number
  - Largest Value
  - max
browser-compat: javascript.builtins.Math.max
---
{{JSRef}}

> Функція **`Math.max()`** повертає найбільше значення з-поміж переданих нуля чи більше вхідних чисел, або {{jsxref("NaN")}}, якщо будь-який з аргументів не є числом чи не може бути приведений до числа.

Reviewer: ```suggestion
Функція **`Math.max()`** повертає найбільше значення з-поміж переданих нуля чи більше вхідних чисел, або {{jsxref("NaN")}}, якщо будь-який з аргументів не є числом і не може бути приведений до числа.
```
Логічніше ж так?

Author: Справедливо. В оригіналі там здається таки `or`, але справді, нахіба нам приводити щось до числа, якщо воно вже число (і тим паче, ловити під час цього `NaN`)

{{EmbedInteractiveExample("pages/js/math-max.html")}}

## Синтаксис

```js
Math.max()
Math.max(value0)
Math.max(value0, value1)
Math.max(value0, value1, ... , valueN)
```

### Параметри

- `value1, value2, ...`
  - : Нуль чи більше чисел, найбільше з яких буде знайдено і повернено.

### Повернене значення

Найбільше з-поміж переданих значень. Якщо один чи більше параметрів не можна перетворити на число, буде повернено {{jsxref("NaN")}}. Якщо жодного параметра передано не було, результат дорівнюватиме \-{{jsxref("Infinity")}}.

## Опис

> Оскільки `max()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.max()`. Не слід звертатись до нього, як до методу створеного вами об'єкту `Math` (`Math` не є конструктором).

Author: ```suggestion
Оскільки `max()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.max()`. Не слід звертатись до нього, як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```

> Оскільки `max()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.max()`. Не слід звертатись до нього, як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).

Reviewer: ```suggestion
Оскільки `max()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.max()`. Не слід звертатись до нього як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```

Якщо жодного аргументу передано не було, результатом буде -{{jsxref("Infinity")}}.

> Якщо принаймні один з аргументів не можна привести до числового вигляду, результат дорівнюватиме {{jsxref("NaN")}}.

Reviewer: ```suggestion
Якщо принаймні один з аргументів не можна привести до числового вигляду, то результатом буде {{jsxref("NaN")}}.
```

## Приклади

### Застосування Math.max()

```js
Math.max(10, 20);   //  20
Math.max(-10, -20); // -10
Math.max(-10, 20);  //  20
```

### Отримання найбільшого з елементів масиву

Для отримання найбільшого з числових елементів масиву можна застосувати {{jsxref("Array.prototype.reduce", "Array.reduce()")}}, шляхом порівняння кожного зі значень:

```js
var arr = [1,2,3];
var max = arr.reduce(function(a, b) {
    return Math.max(a, b);
}, 0);
```

> Наступна функція використовує {{jsxref("Function.prototype.apply()")}} для отримання найбільшого елемента масиву. Виклик `getMaxOfArray([1, 2, 3])` еквівалентний застосуванню `Math.max(1, 2, 3)`, але `getMaxOfArray()` можна використовувати на програмно сконструйованих масивах. Такий підхід сліз застосовувати лише на масивах з відносно невеликою кількістю елементів.

Reviewer: ```suggestion
Наступна функція використовує {{jsxref("Function.prototype.apply()")}} для отримання найбільшого елемента масиву. Виклик `getMaxOfArray([1, 2, 3])` еквівалентний застосуванню `Math.max(1, 2, 3)`, але `getMaxOfArray()` можна використовувати на програмно сконструйованих масивах. Такий підхід слід застосовувати лише на масивах з відносно невеликою кількістю елементів.
```

```js
function getMaxOfArray(numArray) {
  return Math.max.apply(null, numArray);
}
```

> Новий [спред-оператор](/uk/docs/Web/JavaScript/Reference/Operators/Spread_syntax) — це скорочений спосіб запису рішення з `apply` для отримання найбільшого з елементів масиву:

Reviewer: ```suggestion
Новий [оператор розгортання](/uk/docs/Web/JavaScript/Reference/Operators/Spread_syntax) — це скорочений спосіб запису рішення з `apply` для отримання найбільшого з елементів масиву:
```

```js
var arr = [1, 2, 3];
var max = Math.max(...arr);
```

> Однак, як спред-оператор (`...`), так і `apply` або завершаться невдачею, або повернуть некоректне значення, якщо масив містить занадто багато елементів, оскільки вони намагаються передати елементи масиву аргументами до функції. Зверніть увагу на розділ [Застосування `apply` із вбудованими функціями](/uk/docs/Web/JavaScript/Reference/Global_Objects/Function/apply#using_apply_and_built-in_functions) за подробицями. Рішення з `reduce` не має такої проблеми.

Reviewer: ```suggestion
Втім, як оператор розгортання (`...`), так і `apply` – або завершаться невдачею, або повернуть некоректне значення, якщо масив містить занадто багато елементів, оскільки вони намагаються передати елементи масиву аргументами до функції. Зверніть увагу на розділ [Застосування `apply` із вбудованими функціями](/uk/docs/Web/JavaScript/Reference/Global_Objects/Function/apply#using_apply_and_built-in_functions) для отримання подробиць. Рішення з `reduce` не має такої проблеми.
```
